fhir-notification-nats/src/main/java/com/ibm/fhir/notifications/nats/impl/FHIRNotificationNATSPublisher.java

```diff
@@ -0,0 +1,234 @@
+     */
+    public void shutdown() {
+        log.entering(this.getClass().getName(), "shutdown");
+
+        try {
+            if (log.isLoggable(Level.FINE)) {   
+                log.fine("Shutting down NATS publisher for channel: '" + channelName + "'.");
+            }
+            if (sc != null) {
+               sc.close();
+            }
+        } catch (Throwable t) {
+            String msg = "Caught exception shutting down NATS publisher for channel: '" + channelName + "'.";
+            log.log(Level.SEVERE, msg, t);
+            throw new IllegalStateException(msg, t);
+        } finally {
+            log.exiting(this.getClass().getName(), "shutdown");
+        }
+    }
+
+    /**
+     * Publishes an event to NATS.
+     */
+    @Override
+    public void notify(FHIRNotificationEvent event) throws FHIRNotificationException {
+        log.entering(this.getClass().getName(), "notify");
+        String jsonString = null;
+        try {
+            jsonString = FHIRNotificationUtil.toJsonString(event, true);
+
+            if (log.isLoggable(Level.FINE)) { 
+                log.fine("Publishing NATS notification event to channel '" + channelName + "',\nmessage: " + jsonString);
+            }
+            
+            sc.publish("FHIRNotificationEvent", jsonString.getBytes(), acb);
+    
+            if (log.isLoggable(Level.FINE)) {
+                log.fine("Published NATS notification event to channel '" + channelName + "'");
+            }
+        } catch (Throwable e) {
+            String msg = buildNotificationErrorMessage(channelName, (jsonString == null ? "<null>" : jsonString));
+            log.log(Level.SEVERE, msg , e);
+            throw new FHIRNotificationException(msg, e);
+        } finally {
+            log.exiting(this.getClass().getName(), "notify");
+        }
+    }
+    
+    /**
+     * Builds a formatted error message to indicate a notification publication failure.
+     */
+    private String buildNotificationErrorMessage(String channelName, String notificationEvent) {
+        return String.format("NATS publication failure; channel '%s'\nNotification event: %s\n.", channelName, notificationEvent);
+    }
+}
+
+/* 
+* Modified from original NATS documentation @ https://docs.nats.io/developing-with-nats/security/tls
+*
+* This example requires certificates to be in the java keystore format (.jks).
+* To do so openssl is used to generate a pkcs12 file (.p12) from client-cert.pem and client-key.pem.
+* The resulting file is then imported int a java keystore named keystore.jks using keytool which is part of java jdk.
+* keytool is also used to import the CA certificate rootCA.pem into truststore.jks.  
+*/
+class SSLUtils {
+
+    static KeyStore loadKeystore(String path, String password) throws Exception {
+        KeyStore store = KeyStore.getInstance("PKCS12");
+        BufferedInputStream in = new BufferedInputStream(new FileInputStream(path));
```

Comment:
best to use autocloseable

try(BufferedInputStream in = new BufferedInputStream(new FileInputStream(path));){
store.load(in, password.toCharArray());
}